### DISCUSSION TITLE
Wrong usage of copier copy path

### DISCUSSION THREAD

**Discussion by PJ-Schulz** [renovatebot/renovate#30684] [Request Help] [answered]
### What would you like help with?

I think I found a bug

### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

renovate/renovate:38.23.2-full

### Please tell us more about your question or problem

I think I found a bug of the usage of the `copier` command.

In my use case i have a nested repository which contains multiple  .copier-answers.yml files which are not in the project root. Here is a small example of a files structure where my copier app is located at the path `/apps/mytest/copier-test`. 

```
└── apps
    └── mytest
	└── copier-test
		├── .copier-answers.yml
		├── files
		│   ├── file01.yaml
		│   └── file02.yaml
		└── README.md
```

### Manually Update

Now I want to make an update with `copier copy` form 0.2.0 to 0.3.0 where a file called  `apps/mytest/copier-test/files/file03.yaml` will be added.

This is the command i execute when I want to make the update manually without renovate:
```
copier update --skip-answered --defaults --answers-file .copier-answers.yml --vcs-ref 0.3.0 apps/mytest/copier-test/
```

This is the result:
```
└── apps
    └── mytest
	└── copier-test
		├── .copier-answers.yml
		├── files
		│   ├── file01.yaml
		│   ├── file02.yaml
		│   └── file03.yaml
		└── README.md
```

### Now what renovate does

I have investigated that renovate will execute this command:
```
copier update --skip-answered --defaults --answers-file apps/mytest/copier-test/.copier-answers.yml --vcs-ref 0.3.0 
```
Which leads to this file structure:
```
├── apps
│   └── mytest
│       └── copier-test
│           ├── .copier-answers.yml
│	    ├── files
│           │   ├── file01.yaml
│           │   └── file02.yaml
│           └── README.md
├── files
│    └── file03.yaml
└── .copier-answers.yml
```

You can see that the new file is placed at the wrong path.
* Expected: `apps/mytest/copier-test/files/file03.yaml` 
* Real: `files/file03.yaml` 

### What happend?

The used copier command from renovate does not include the destination path. It has only the path where the answer file is located. So because there is no destination path, copier will place the new loaded file based at the root path or the repo.

### How to fix?:

The function for building the command must be changed:

https://github.com/renovatebot/renovate/blob/7737b6cf01c4dbc23a7a686637aa46b7f16833eb/lib/modules/manager/copier/artifacts.ts#L20-L37

The parameter `packageFileName` is in for example: `apps/mytest/copier-test/.copier-answers.yml`. This string must be spitted at the last part. The first chunk can be used as destination at the end of the command and the last chunk as answer file. So the resulting command should look like this:

```
copier update --skip-answered --defaults --answers-file .copier-answers.yml --vcs-ref 0.3.0 apps/mytest/copier-test/
```

---
```
# OLD
copier ... --answers-file apps/mytest/copier-test/.copier-answers.yml --vcs-ref 0.3.0 
# NEW
copier ... --answers-file .copier-answers.yml --vcs-ref 0.3.0 apps/mytest/copier-test/
```

**Comment by rarkins** [comment]
Please submit a PR fixing it or create a reproduction repo so that a bug issue can be raised

**Reply by PJ-Schulz** [reply, marked as answer]
I opened a PR here: #30708
